includes/plugin.php

```diff
@@ -841,6 +842,8 @@ public function __get( $property ) {
 	private function __construct() {
 		$this->register_autoloader();
 
+		Container::set_instance();
```

Comment:
Maybe add some `register` method? and do not connect the registration process to the building of an instance part?